java/common/src/main/scala/org/typelevel/otel4s/java/TextMapPropagatorImpl.scala

```diff
@@ -39,4 +39,18 @@ private[java] class TextMapPropagatorImpl[F[_]: Sync](
     Sync[F].delay(
       jPropagator.inject(toJContext(ctx), carrier, fromTextMapSetter)
     )
+
+  def injected[A](ctx: Vault, carrier: A)(implicit
+      injector: TextMapUpdater[A]
+  ): A = {
+    var injectedCarrier = carrier
```

Comment:
incidentally, this gets transparently boxed as an [`ObjectRef`](https://github.com/scala/scala/blob/v2.13.10/src/library/scala/runtime/ObjectRef.java) by the compiler anyway

Reply:
Interesting.  I figured there was already something like `Box`, but I'm also surprised anything is added here.  Do you know where I can read more?